### ISSUE TITLE
Accessibility: make it possible for screen readers to report state change from not faved to faved

### ISSUE THREAD

**Issue by mem** [mastodon/mastodon#24661]
### Pitch

Users of screen readers would benefit from annotations in the HTML elements that allow them to be notified when a toggle like reblog/fav has been clicked or not. Other users have the possibility to see visual changes in the corresponding icons (e.g. the fav star spins and becomes yellow when clicked).

### Motivation

I'm reporting this on behalf of a user with low vision trying to use a screen reader to interact with the Mastodon web client.

They say it's difficult to interact with the "fav" icon because they don't know whether it's been clicked or not. The screen reader doesn't have enough information to let them know the state of the icon.

https://mastodon.cr/@Wilmer/110261860179271829

I have no experience in accessibility, but my cursory reading of the WAI-ARIA document suggest this is possible using the "switch" role:

https://www.w3.org/TR/wai-aria-1.1/#switch
